src/pkcs11/framework-pkcs15.c

```diff
@@ -1595,10 +1608,27 @@ pkcs15_create_tokens(struct sc_pkcs11_card *p11card, struct sc_app_info *app_inf
 	}
 	sc_log(context, "Use FW data with index %i; fw_data->p15_card %p", idx, fw_data->p15_card);
 
+	conf_block = sc_get_conf_block(p11card->card->ctx, "framework", "pkcs15", 1);
+	if (conf_block && app_info)   {
+		scconf_block **blocks = NULL;
+		char str_path[SC_MAX_AID_STRING_SIZE];
+
+		memset(str_path, 0, sizeof(str_path));
+		sc_bin_to_hex(app_info->path.value, app_info->path.len, str_path, sizeof(str_path), 0);
+		blocks = scconf_find_blocks(p11card->card->ctx->conf, conf_block, "application", str_path);
+		if (blocks)   {
+			if (blocks[0]) {
+				user_pin = (char *)scconf_get_str(blocks[0], "user_pin", NULL);
```

Comment:
As it is now, for our eID a call to `_get_auth_object_by_name("UserPIN")` returns a "CardCAN" PIN object (the first one in a AODF) and a call to `_get_auth_object_by_name("SignPIN")` returns "NormPIN" (the second one in a AODF), but it should return "NormPIN" and "Sig PIN" respectively.

"UserPIN/SignPIN" as they are now used in `_get_auth_object_by_name()` are not a reference to a PIN object name/label, but rather specify how the "suitable" PIN is selected/returned. For example, currently when "UserPIN" is specified it returns the first "global" PIN (in case of our eID card that is "CardCAN" and not a PIN named "UserPIN" (that is required). With this patch it is possible to explicitly say, I want to use PIN named "user_pin=UserPIN" as the "User PIN" in PKCS#11 context and not the one library would return ("Card CAN").

Reply:
I understand. Please add this to the documentation then.

Reply:
Already done.